examples/Makefile

```diff
@@ -155,116 +166,151 @@ IMAGE_XDP_BC ?= quay.io/bpfman-bytecode/go-xdp-counter:latest
 IMAGE_XDP_US ?= quay.io/bpfman-userspace/go-xdp-counter:latest
 IMAGE_KP_BC ?= quay.io/bpfman-bytecode/go-kprobe-counter:latest
 IMAGE_KP_US ?= quay.io/bpfman-userspace/go-kprobe-counter:latest
+IMAGE_UP_BC ?= quay.io/bpfman-bytecode/go-uprobe-counter:latest
+IMAGE_UP_US ?= quay.io/bpfman-userspace/go-uprobe-counter:latest
+IMAGE_GT_US ?= quay.io/bpfman-userspace/go-target:latest
 KUST_DIR=default
 KIND_CLUSTER_NAME ?= bpfman-deployment
 
-.PHONY: deploy-tc
-deploy-tc: kustomize ## Deploy go-tc-counter to the cluster specified in ~/.kube/config.
+
+.PHONY: deploy-prog
+deploy-prog: kustomize
 ifndef TAG
-	sed 's@URL_BC@$(IMAGE_TC_BC)@' config/default/go-tc-counter/patch.yaml.env > config/default/go-tc-counter/patch.yaml
-	cd config/default/go-tc-counter && $(KUSTOMIZE) edit set image quay.io/bpfman-userspace/go-tc-counter=${IMAGE_TC_US}
+	sed 's@URL_BC@$(IMAGE_BC)@' config/default/$(CONFIG_DIR)/patch.yaml.env > config/default/$(CONFIG_DIR)/patch.yaml
+	cd config/default/$(CONFIG_DIR) && $(KUSTOMIZE) edit set image quay.io/bpfman-userspace/$(PROG_NAME)=${IMAGE_US}
 else
 	$(eval KUST_DIR=$(TAG))
 endif
-	$(KUSTOMIZE) build config/$(KUST_DIR)/go-tc-counter | kubectl apply -f -
+	$(KUSTOMIZE) build config/$(KUST_DIR)/$(CONFIG_DIR) | kubectl apply -f -
 
-.PHONY: undeploy-tc
-undeploy-tc: ## Undeploy go-tc-counter from the cluster specified in ~/.kube/config.
+
+.PHONY: undeploy-prog
+undeploy-prog:
 ifndef TAG
-	sed 's@URL_BC@$(IMAGE_TC_BC)@' config/default/go-tc-counter/patch.yaml.env > config/default/go-tc-counter/patch.yaml
-	cd config/default/go-tc-counter && $(KUSTOMIZE) edit set image quay.io/bpfman-userspace/go-tc-counter=${IMAGE_TC_US}
+	sed 's@URL_BC@$(IMAGE_BC)@' config/default/$(CONFIG_DIR)/patch.yaml.env > config/default/$(CONFIG_DIR)/patch.yaml
+	cd config/default/$(CONFIG_DIR) && $(KUSTOMIZE) edit set image quay.io/bpfman-userspace/$(PROG_NAME)=${IMAGE_US}
 else
 	$(eval KUST_DIR=$(TAG))
 endif
-	$(KUSTOMIZE) build config/$(KUST_DIR)/go-tc-counter | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
+	$(KUSTOMIZE) build config/$(KUST_DIR)/$(CONFIG_DIR) | kubectl delete --ignore-not-found=$(ignore-not-found) -f -; \
+
+
+.PHONY: deploy-tc
+deploy-tc: PROG_NAME=go-tc-counter
+deploy-tc: CONFIG_DIR=$(PROG_NAME)
+deploy-tc: IMAGE_BC=$(IMAGE_TC_BC)
+deploy-tc: IMAGE_US=$(IMAGE_TC_US)
+deploy-tc: deploy-prog ## Deploy go-tc-counter to the cluster specified in ~/.kube/config.
```

Comment:
Don't duplicate this target just put the env vars at the top of the makefile

OR just make it one

deploy-tc:
    PROG_NAME=go-tc-counter
    CONFIG_DIR=prog_name

Reply:
Unless I'm misunderstanding, I don't think what you are suggesting will work.  I tried it.  I hate Makefiles :-).

To separate the instructions from the variables, I needed to use [target-specific variables](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html).  This allows me to define a separate set of variables for each target.  In theory, we could use the same set of variables for the deploy and undeploy versions for a given program, but then we'd need a separate variable to be passed in to describe the action.  E.g., make `tc-prog ACTION=DEPLOY`.  I think this is more clumsy for the user.

Reply:
Also, the multiple lines prefixed by `deploy-tc:` aren't duplicating the target.  That's how you defined target-specific variables in a Makefile.  It tells `make` that these variables only apply to the given target.